flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClientFactory.java

```diff
@@ -81,6 +91,57 @@ private YarnClusterDescriptor getClusterDescriptor(Configuration configuration)
 
         yarnClient.init(yarnConfiguration);
         yarnClient.start();
+        if (configuration.contains(YarnConfigOptions.APPLICATION_NAME)) {
+            try {
+                Set<String> applicationTypes = Sets.newHashSet();
+                applicationTypes.add("Apache Flink");
```

Comment:
I think the application type can be something different than `"Apache Flink"`. See `YarnConfigOptions.APPLICATION_TYPE`.

Reply:
Yes, it's a wrong code, i will fix it.